src/components/PlayersComponent.tsx

```diff
@@ -1,25 +1,22 @@
 import * as React from "react";
 import { useState } from "react";
-
 import { Button, Pane, TextInput } from "evergreen-ui";
 import { Quiz, AppMethods } from "../store/types";
-
-import { useHistory } from "react-router";
+import { PlayQuizComponent } from "./PlayQuizComponent";
 
 // Modifies the players list
 export const PlayersComponent = (props: {
   quiz: Quiz;
   index: string;
-  addPlayers: AppMethods["addPlayers"];
-  editPlayers: AppMethods["editPlayers"];
+  setPlayers: AppMethods["setPlayers"];
+  onClose: () => void;
```

Comment:
Maybe pass a single prop: `onFinish: ( players: string[] ) => void` and in the parent comp you do something like: `... onFinish={ ( players ) => setPlayers(players); setShowPlayers(false) }`.

Like this the players component doesn't have to care about closing itself.